Source/Core/TrustedServers.js

```diff
@@ -0,0 +1,148 @@
+/*global define*/
+define([
+    './defined',
+    './defineProperties',
+    './DeveloperError',
+    '../ThirdParty/Uri'
+], function(
+    defined,
+    defineProperties,
+    DeveloperError,
+    Uri) {
+    'use strict';
+    
+    /**
+     * A singleton that contains all of the servers that are trusted. Credentials will be sent with
+     * any requests to these servers.
+     *
+     * @exports TrustedServers
+     *
+     * @see {@link http://www.w3.org/TR/cors/|Cross-Origin Resource Sharing}
+     */
+    var TrustedServers = {};
+    var _servers = {};
+
+    /**
+     * Adds a trusted server to the registry
+     *
+     * @param {String} domain The domain to be added.
```

Comment:
Being somewhat pedantic, the spec uses the terms "host" and "port".

https://tools.ietf.org/html/rfc3986#section-3.2